### ISSUE TITLE
Introspection query sets type name to null

### ISSUE THREAD

**Issue by gauravtiwari** [rmosolgo/graphql-ruby#383]
Hey @rmosolgo 

Getting following error when using 1.2 version (`graphql-js`) with Relay and also in GraphiQL

```js
Syntax Error GraphQL (1:1) Unexpected Name "null"

1: null
   ^
```

It looks like in Schema it sets the type name to `null`

```json
 {
                  "name": "id",
                  "description": null,
                  "type": {
                    "kind": "NON_NULL",
                    "name": null,
                    "ofType": {
                      "kind": "SCALAR",
                      "name": "String",
                      "ofType": null
         }
   },
```

**Comment by gauravtiwari** [comment]
Looks like this PR made the change in 1.1.0 (https://github.com/rmosolgo/graphql-ruby/pull/355)

**Comment by rmosolgo** [comment]
This seems to be in keeping with the spec: 

![image](https://cloud.githubusercontent.com/assets/2231765/20085536/8be64d1c-a51e-11e6-921e-051819721c8a.png)


(https://facebook.github.io/graphql/#sec-Type-Kinds.Non-Null)

And, it seems that [graphql-swapi](http://graphql-swapi.parseapp.com/) returns `null` name also: 

![image](https://cloud.githubusercontent.com/assets/2231765/20085609/1c256ba6-a51f-11e6-8317-e351fc3ac625.png)

Can you help me understand where that syntax error comes from?

**Comment by rmosolgo** [comment]
Here's a non-null response on swapi: 

![image](https://cloud.githubusercontent.com/assets/2231765/20085667/6d6df0d2-a51f-11e6-9b6e-ed7f611640f9.png)

**Comment by gauravtiwari** [comment]
@rmosolgo Yup that makes sense. The error comes when validating schema with GraphQL(javascript) on client side. Relay uses it under the hood to parse the schema. Perhaps, this needs to be fixed on JS version of GraphQL.

**Comment by jamesmacaulay** [comment]
@gauravtiwari that error comes from `graphql-js`'s parser, which as far as I know would not be used to process a JSON response of an introspection query. Here are tests in `graphql-js` which show that kind of error when parsing queries, however:

https://github.com/graphql/graphql-js/blob/c23f5782dee81b0a59a6a6bb2bcca9da247570e7/src/language/__tests__/parser-test.js#L54-L56
https://github.com/graphql/graphql-js/blob/c23f5782dee81b0a59a6a6bb2bcca9da247570e7/src/language/__tests__/parser-test.js#L82-L86

Is it possible that this error is a result of a query value somewhere in the JS code which is `null`, and then coerced to a string `"null"`, and then attempted to be parsed by the query parser?

```javascript
Syntax Error GraphQL (1:1) Unexpected Name "null"

1: null
   ^
```

...it looks like `"null"` is the entire body of the string being parsed there.

**Comment by gauravtiwari** [comment]
@jamesmacaulay looks like this is fixed in `graphql-js 0.8.0`. I am going to close this one and update babel relay plugin to use new version of GraphQL.
